sematic/plugins/building/docker.py

```diff
@@ -0,0 +1,472 @@
+
+        return build_image_uri
+
+    def _launch(self, target: str, image_uri: ImageURI) -> None:
+        """
+        Launches the specified user code target, using the specified image.
+        """
+        sys.path.append(os.getcwd())
+        # TODO: revert this overwrite after finishing execution
+        #  promote the `environment_variables` testing fixture to a utility
+        os.environ[CONTAINER_IMAGE_ENV_VAR] = repr(image_uri)
+
+        logger.info("Launching target: '%s'", target)
+
+        runpy.run_path(path_name=target, run_name="__main__")
+
+        logger.info("Finished launching target: '%s'", target)
+
+
+def _get_build_config(script_path: str) -> BuildConfig:
+    """
+    Handles the entire instantiation of the definite `BuildConfig` object for the
+    specified script.
+
+    Raises
+    ------
+    BuildConfigurationError:
+        There was an error when loading or validating the specified build configuration.
+    """
+    build_config_files = _find_build_config_files(script_path=script_path)
+    logger.info(
+        "Script '%s' has these corresponding build files: %s",
+        script_path,
+        build_config_files,
+    )
+
+    # TODO: override several config files, env vars, and cli arguments
+    if len(build_config_files) == 0:
+        raise BuildConfigurationError(
+            f"Unable to find any build files corresponding to script '{script_path}'! "
+            f"Please see TODO for build configuration details!"
+        )
+
+    return BuildConfig.load_build_config_file(build_config_files[0])
+
+
+def _find_build_config_files(script_path: str) -> List[str]:
+    """
+    Searches for build configuration files that correspond to the specified script file,
+    and returns a list containing their paths.
+    """
+    # TODO: also load `build.yaml` files on the file hierarchy between the cwd and the
+    #  script directory
+    root = os.path.splitext(script_path)[0]
+    file_candidate = f"{root}.yaml"
+
+    return [file_candidate] if os.path.isfile(file_candidate) else []
+
+
+def _get_image_uri(build_config: BuildConfig, target: str) -> ImageURI:
+    """
+    Returns the container image URI to use, according to the build configuration.
+
+    Raises
+    ------
+    BuildError:
+        There was an error when executing the specified build script.
+    BuildConfigurationError:
+        There was an error when validating the specified build configuration.
+    """
+    if build_config.base_uri is not None:
+        # TODO: honor configurations in build_config.build
+        return build_config.base_uri
+
+    if build_config.image_script is None:
+        # appease mypy
+        raise BuildConfigurationError(
+            "Exactly one of `image_script` and `base_uri` must be specified!"
+        )
+
+    try:
+        script_dir, script_file = os.path.split(build_config.image_script)
+        if script_dir == "":
+            script_dir = None  # type: ignore
```

Comment:
Maybe you can type `script_dir` as `Optional[str]` to avoid the `# type: ignore`?